cmake/app/boilerplate.cmake

```diff
@@ -335,6 +335,14 @@ foreach(root ${BOARD_ROOT})
         ${shield_dir}/${s_dir}/dts_fixup.h
         )
 
+      # search for shield dts root directory
+      if(EXISTS ${shield_dir}/${s_dir}/dts AND IS_DIRECTORY ${shield_dir}/${s_dir}/dts)
```

Comment:
The test for existence of the dts folder in a DTS_ROOT is also done here:
https://github.com/zephyrproject-rtos/zephyr/blob/09e9736b4b1a3e8c4bb572a5fe544956800990b9/cmake/dts.cmake#L90-L97
https://github.com/zephyrproject-rtos/zephyr/blob/09e9736b4b1a3e8c4bb572a5fe544956800990b9/cmake/dts.cmake#L107-L109

so we don't need to test the existence of the `dts` folder here.

Removing `EXISTS` and `IS_DIRECTORY` here ensures that adding shield dir to the list of dts roots behaves the same as any other dts root (board, application, etc.).
It also has the benefit that we only need to maintain those checks at a single location.

---

To simplify, and try to keep names aligned, then maybe we should create a list containing shield dirs, like:
```
list(APPEND SHIELD_DIRS ${shield_dir}/${s_dir})
```
The order in that list should of course match the order in the `SHIELD_LIST` so they can be used together.

Having such a list would be similar the the `BOARD_DIR` variable, and using the name `SHIELD_DIRS` indicates a variable accessible in whole build system (I know we have not been good at keeping variable names consistent).

The `SHIELD_DIRS` can then be added to `DTS_ROOT` as other default roots, but will also be available for other uses if needed.

Reply:
@tejlmand I see the idea behind having a `SHIELD_LIST` variable, but shouldn't its order follow `SHIELD` and not `SHIELD_LIST`? This is how `shield_dts_files`, `shield_dts_fixups`, and `shield_conf_files` are sorted today, and provides the user the possibility of controlling the evaluation order.

Reply:
> This is how shield_dts_files, shield_dts_fixups, and shield_conf_files are sorted today, and provides the user the possibility of controlling the evaluation order.

good point, let's not do any changes in this PR that could impact this order.